ComfortZoneApp/app/src/main/java/com/example/comfortzone/fragments/FlightFragment.java

```diff
@@ -228,22 +235,24 @@ public void onButtonChecked(MaterialButtonToggleGroup group, int checkedId, bool
     }
 
     private void updateViewsList(List<WeatherData> cityList) {
-        CityListViewFragment cityListViewFragment = (CityListViewFragment) fragmentManager.findFragmentByTag(TAG_LIST_VIEW);
-        cityListViewFragment.onCityListUpdated(cityList);
+        if (cityListViewFragment != null) {
+            cityListViewFragment.onCityListUpdated(cityList);
+        }
+        if (mapFragment != null) {
+            mapFragment.onCityListUpdated(cityList);
```

Comment:
i see, so you are updating both even when it's not on screen? I think that's fine, because there's an edge case where if you switch to the other, we want to make sure the list is up to date.